xbmc/platform/win32/WinMain.cpp

```diff
@@ -88,6 +89,31 @@ static std::shared_ptr<CAppParams> ParseCommandLine()
   return appParamParser.GetAppParams();
 }
 
+static bool CheckAndSetFileFlag(bool usePlatformDirectories, HANDLE& handle)
+{
+  const std::string file = CWIN32Util::GetProfilePath(usePlatformDirectories) + "\\.running";
+
+  using KODI::PLATFORM::WINDOWS::ToW;
+  handle =
+      CreateFile(ToW(file).c_str(), 0, 0, nullptr, CREATE_NEW, FILE_FLAG_DELETE_ON_CLOSE, NULL);
+
+  if (handle == INVALID_HANDLE_VALUE && GetLastError() == ERROR_FILE_EXISTS)
+    return true;
+
+  return false;
+}
+
+static bool CheckAndSetMutex(HANDLE& handle)
```

Comment:
May I suggest to put this function into an unnamed namespace instead of tagging it static?